src/clipboard_handlers/merge_clipboard.ts

```diff
@@ -18,12 +18,15 @@ export class MergeClipboardHandler extends AbstractCellClipboardHandler<
   ClipboardContent,
   Maybe<Merge>
 > {
-  copy(data: ClipboardCellData): ClipboardContent | undefined {
+  copy(data: ClipboardCellData, isCutOperation?: boolean): ClipboardContent | undefined {
     const sheetId = this.getters.getActiveSheetId();
     const { rowsIndexes, columnsIndexes } = data;
     const merges: Maybe<Merge>[][] = [];
 
     for (const row of rowsIndexes) {
+      if (!isCutOperation && this.getters.isRowFiltered(sheetId, row)) {
```

Comment:
merges with half their rows filtered are still copied in full. Not sure if it's/what the correct behaviour is

Reply:
Probably not the best solution (but can't figure what the best solution should be), but we have the same behavior as GSheet